app_utils/chat.py

```diff
@@ -0,0 +1,82 @@
+import json
+import re
+from typing import Dict, Any
+
+import requests
+import streamlit as st
+from snowflake.connector import SnowflakeConnection
+
+API_ENDPOINT = "https://{HOST}/api/v2/cortex/analyst/message"
+
+
+@st.cache_data(ttl=60, show_spinner=False)
+def send_message(
```

Comment:
I just moved `send_message` out of `iteration.py` and into `chat.py` so that we could share it with the validation code.

Most of the code is the same except for how we handle errors; we throw a `ValueError` and leave exception handling up to the caller instead of outputting a Streamlit error component.